models/maat/R/model2netcdf.MAAT.R

```diff
@@ -50,6 +31,37 @@ model2netcdf.MAAT <- function(rundir, outdir, sitelat = -999, sitelon = -999, st
   # setup constants
   day_secs <- udunits2::ud.convert(1, "day", "seconds")
   
+  # setup helper function
+  var_update <- function(data, out, oldname, newname, oldunits, newunits=NULL, missval=-999, longname, ncdims) {
+    # ifelse is no longer working as expected, so now we have this function to deal with any Inf values
+    f_sort <- function(s) {
+      if (is.infinite(s)) -999
+      else try(PEcAn.utils::misc.convert(s, oldunits, newunits), silent = TRUE)
+    }
+    ## define variable
+    if(is.null(newunits)) newunits = oldunits
+    newvar <- ncdf4::ncvar_def(name = newname, units = newunits, dim = ncdims, missval=missval, longname=longname)
+    ## convert data
+    dat <- data
+    if (newname %in% c("Year","FracJulianDay")) {
+      PEcAn.logger::logger.info(paste0("Skipping conversion for: ", newname))
+      dat.new <- dat
+    } else {
+      dat.new <- apply(as.matrix(dat,length(dat),1),1,f_sort)
+    }
+    ## prep for writing
+    if(is.null(out)) {
+      out <- list(var <- list(),dat <- list())
```

Comment:
Unless you meant to do a global assignment...

```suggestion
      out <- list(var = list(), dat = list())
```